app/controllers/api/v1/notes_controller.rb

```diff
@@ -0,0 +1,45 @@
+module Api
+  module V1
+    class NotesController < ApplicationController
+      def index
+        type = params[:type]
+        page_size = params[:page_size]
+        max_page_size = 100
+
+        if page_size.to_i > max_page_size
+          return render json: { error: "page_size is too long, max allowed is #{max_page_size}" }, status: :bad_request
+        end
+
+        if type && !Note.types.keys.include?(type)
+          return render json: { error: "invalid type #{type}" }, status: :unprocessable_entity
+        end
+        render json: notes_filtered, status: :ok, each_serializer: IndexNoteSerializer
+      end
+
+      def show
+        render json: show_note, status: :ok, serializer: ShowNoteSerializer
+      end
+
+      private
+
+      def notes
+        # TODO: Use current_user to get notes associated to the user
+        # current_user.notes
+        Note.all
+      end
+
+      def notes_filtered
+        order, page, page_size = params.values_at(:order, :page, :page_size)
+        notes.where(filtering_params).order(created_at: order || :desc).page(page).per(page_size)
```

Comment:
No solemos realizar las queries acá en el controller, usamos algo que se llama scope que se puede definir en el modelo, si querés podes revisarlo en la docu: 🔗 https://www.rubyguides.com/2019/10/scopes-in-ruby-on-rails/

Reply:
Perfectoo, ahí usé un scope y quedó mucho mejor. Gracias!